pyanaconda/packaging/__init__.py

```diff
@@ -374,18 +376,55 @@ def _getTreeInfo(self, url, proxy_url, sslverify):
                 log.info("Failed to parse proxy for _getTreeInfo %s: %s",
                          proxy_url, e)
 
-        response = None
+        # Retry treeinfo downloads with a progressively longer pause,
+        # so NetworkManager have a chance setup a network and we have
+        # full connectivity before trying to download things. (#1292613)
+        xdelay = xprogressive_delay()
+        response = res_1 = res_2 = None
```

Comment:
I don't like this construct. What about having this as a list of responses? Altogether this code seems to be really complex. Maybe some of it could go into a function?